src/suites/cts/validation_test.ts

```diff
@@ -0,0 +1,233 @@
+import GLSL from '../../tools/glsl.macro.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export class ValidationTest extends GPUTest {
+  uniformBuffer: GPUBuffer = undefined!;
+  storageBuffer: GPUBuffer = undefined!;
+  sampler: GPUSampler = undefined!;
+  sampledTexture: GPUTexture = undefined!;
+
+  async init(): Promise<void> {
+    await super.init();
+
+    this.uniformBuffer = this.device.createBuffer({
```

Comment:
These default values are somewhat arbitrary so I don't think they should be in the fixture for all of the validation tests. Instead I suggest adding them to the fixture for bind group layouts specifically.

Reply:
Maybe these don't even need to be on the fixture - could probably just have standalone helper functions in the bind group test file that create new buffers of various types